editor/test/internal/node_test.clj

```diff
@@ -611,30 +606,551 @@
         (is (= @production-count 1))))))
 
 (g/defnode CachedDependencyTestNode
+  (property property g/Any)
   (input regular-input g/Any)
   (input array-input g/Any :array)
-  (output internal-output g/Any :cached (g/fnk [_node-id] _node-id))
-  (output evaluated-output g/Any :cached (g/fnk [regular-input array-input internal-output] [regular-input array-input internal-output])))
+
+  ;; Dependent on property.
+  (output uncached-output<=property g/Any (g/fnk [property] property))
+  (output cached-output<=property g/Any :cached (g/fnk [property] property))
+  (output uncached-secondary-output<=uncached-output<=property g/Any (g/fnk [uncached-output<=property] uncached-output<=property))
+  (output cached-secondary-output<=uncached-output<=property g/Any :cached (g/fnk [uncached-output<=property] uncached-output<=property))
+  (output uncached-secondary-output<=cached-output<=property g/Any (g/fnk [cached-output<=property] cached-output<=property))
+  (output cached-secondary-output<=cached-output<=property g/Any :cached (g/fnk [cached-output<=property] cached-output<=property))
+
+  ;; Dependent on regular-input.
+  (output uncached-output<=regular-input g/Any (g/fnk [regular-input] regular-input))
+  (output cached-output<=regular-input g/Any :cached (g/fnk [regular-input] regular-input))
+  (output uncached-secondary-output<=uncached-output<=regular-input g/Any (g/fnk [uncached-output<=regular-input] uncached-output<=regular-input))
+  (output cached-secondary-output<=uncached-output<=regular-input g/Any :cached (g/fnk [uncached-output<=regular-input] uncached-output<=regular-input))
+  (output uncached-secondary-output<=cached-output<=regular-input g/Any (g/fnk [cached-output<=regular-input] cached-output<=regular-input))
+  (output cached-secondary-output<=cached-output<=regular-input g/Any :cached (g/fnk [cached-output<=regular-input] cached-output<=regular-input))
+
+  ;; Dependent on array-input.
+  (output uncached-output<=array-input g/Any (g/fnk [array-input] array-input))
+  (output cached-output<=array-input g/Any :cached (g/fnk [array-input] array-input))
+  (output uncached-secondary-output<=uncached-output<=array-input g/Any (g/fnk [uncached-output<=array-input] uncached-output<=array-input))
+  (output cached-secondary-output<=uncached-output<=array-input g/Any :cached (g/fnk [uncached-output<=array-input] uncached-output<=array-input))
+  (output uncached-secondary-output<=cached-output<=array-input g/Any (g/fnk [cached-output<=array-input] cached-output<=array-input))
+  (output cached-secondary-output<=cached-output<=array-input g/Any :cached (g/fnk [cached-output<=array-input] cached-output<=array-input)))
 
 (deftest dependency-caching
+  (letfn [(cacheable-property-dependent-endpoints [node-id]
```

Comment:
Fleshed out this existing test to cover more editing scenarios. This was done during the discovery process. The new checks don't actually fail if I remove the fix in `transaction.clj`, but I like the added coverage.